### ISSUE TITLE
Nuxt 3 A/B Test Variant Rendered Client Side

### ISSUE THREAD

**Issue by KikoTs** [BuilderIO/builder#2904]
**Describe the bug**
In a split test with two variants, the chosen variant is inconsistently rendered on the client side. This results in the main content, initially rendered on the server side, being replaced on the client side by the selected variant. Additionally, there is a noticeable flash during the rotation of variants, impacting the user experience.

**To Reproduce**
1. Create a page in the builder and define two variants.
2. Render the page using the Nuxt Catch All method.
3. Observe the issue where the page exhibits a flash and the variant rendering appears inconsistent.

**Expected behavior**
The expectation is for the server to instantly render the chosen variant, eliminating the flash and ensuring consistent display.

**Screenshots**
[Video of the issue](https://drive.google.com/file/d/1Fv89AV1KN2wmk2HdvXBe4iJ3ZwCh3EcH/view)

**Additional Context and Code Snippet**:
- Framework: NuxtJS 3
- Library: @builder.io/sdk-vue/vue3

I suspect the issue might be related to the "RenderContent" component. Below is the code snippet used to fetch and render the sections:

```vue
<script setup>
const { data: content } = await useAsyncData('content', () =>  {
  const response = getContent({
  model: 'page',
  apiKey: BUILDER_PUBLIC_API_KEY,
  userAttributes: {
    urlPath: `/${props.slug[0]}`, // page url
  },
})
return response
})
</script>

<template>
  <div v-if="content || isPreviewing()">
    <RenderContent
      model="page" :content="content" :api-key="BUILDER_PUBLIC_API_KEY"
      :custom-components="REGISTERED_COMPONENTS">
    </RenderContent>
  </div>
</template>
```

I am looking for insights or suggestions on how to ensure seamless variant rendering without the client-side flash.

**Comment by samijaber** [comment]
Can you confirm the SDK version you're on? 0.7.4 includes multiple SSR A/B testing fixes: https://github.com/BuilderIO/builder/blob/main/packages/sdks/output/vue/CHANGELOG.md#patch-changes-1

Otherwise, could you share a Content URL (`builder.io/content/CONTENT_ID`), or staging or production URL highlighting this issue?

**Comment by KikoTs** [comment]
I was using the SDK Version 0.8.0 as of the writing of the issue, my content url is 
[https://builder.io/content/0916067cdfe64d7296e9de5becf4853f](https://builder.io/content/0916067cdfe64d7296e9de5becf4853f)

And a production url where the issue can be seen is https://sale.lifeboostcoffee.com/healthy-coffee-23

Note: as a temporary patch there is implemented fader to hide the page flashing

**Comment by samijaber** [comment]
@KikoTs We just published v1.0.18 which includes a fix for A/B test flickering:

https://github.com/BuilderIO/builder/releases/tag/%40builder.io%2Fsdk-vue%401.0.18

Please test it out when you have a chance and let me know if it resolves your issues.

**Comment by KikoTs** [comment]
Amazing, works like a charm!
Thanks for resolving the issue!🙌
